### PR TITLE
fix:default value for PhoneInput set to text which is invalid(#34594)

### PR DESCRIPTION
**Pull request by Harshithazemoso**
Fixes #`34594`  
_or_  
Fixes `https://github.com/appsmithorg/appsmith/issues/34594`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

**SCREENSHOTS AFTER SOLVING THE ISSUE**:
No issue when an number is entered 
Screenshot from 2024-07-02 17-28-02
Shows error when a text is entered
Screenshot from 2024-07-02 17-26-20

**VIDEO**
[VIDEO:CYPRESS TESTING](https://www.loom.com/share/d82acfdfe1f44d808d8008cc46958eb1?sid=7bddd43b-d728-482b-93b3-d51965e76f14)

## Summary by CodeRabbit

- **New Features**
  - Enhanced Phone Input widget with improved default value validation and error handling.

- **Tests**
  - Added test cases for Phone Input widget to ensure proper default value behavior and error message handling.

- **Bug Fixes**
  - Updated error message for Phone Input widget to clearly indicate when a value must be a number.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/PhoneInput/PhoneInput_DefaultValue_spec.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    agHelper,
+    locators,
+    entityExplorer,
+    propPane,
+  } from "../../../../../support/Objects/ObjectsCore";
+  const commonlocators = require("../../../../../locators/commonlocators.json");
+  
+  describe(
+    "Phone Input widget Tests",
+    { tags: ["@tag.Widget", "@tag.PhoneInput"] },
```
Comment (ApekshaBhosale) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/PhoneInput/PhoneInput_DefaultValue_spec.ts:11]: 1. Add description to understand this spec more
2. remove before and make everything as one test
3. Don't use string to validate messages. use variable instead from respective files 

@sagar-qa007

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/PhoneInput/PhoneInput_DefaultValue_spec.ts (hunk 1 above)
Comment (Harshithazemoso) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/PhoneInput/PhoneInput_DefaultValue_spec.ts:11]: > 1. Add description to understand this spec more
> 2. remove before and make everything as one test
> 3. Don't use string to validate messages. use variable instead from respective files
> 
> @sagar-qa007

hello @ApekshaBhosale can you please be more specific about the third one ,are you saying about  this,
 'agHelper.UpdateCodeInput(propPane._propertyContentDefaultValueField, '9872139');',
 and 
  'agHelper.VerifyEvaluatedErrorMessage('This value must be number');'

Comment (ApekshaBhosale) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/PhoneInput/PhoneInput_DefaultValue_spec.ts:11]: i am talking about `'agHelper.VerifyEvaluatedErrorMessage('This value must be number');'`

Comment (Harshithazemoso) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/PhoneInput/PhoneInput_DefaultValue_spec.ts:11]: Everywhere they are using the string @ApekshaBhosale , I think its not declared anywhere ,I couldnt find any file where its declared

**Comment by Harshithazemoso** [comment]
hello @ApekshaBhosale made changes as suggested ,can you review it now

**Pull request closed without merging**